tests/acl/test_stress_acl.py

```diff
@@ -1,50 +1,126 @@
 import logging
+import re
 import pytest
+import ptf.testutils as testutils
+from ptf import mask, packet
+from collections import namedtuple
 from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
 from tests.common.dualtor.mux_simulator_control import toggle_all_simulator_ports_to_rand_selected_tor  # noqa F401
 from tests.common.utilities import wait
+from tests.route.test_route_flap import get_ip_route_info, get_route_prefix_len
 
 pytestmark = [
-    pytest.mark.topology("t0", "t1", "m0", "mx"),
-    pytest.mark.device_type('vs')
+    pytest.mark.topology("t0", "t1", "m0", "mx")
 ]
 
 logger = logging.getLogger(__name__)
 
 LOOP_TIMES_LEVEL_MAP = {
     'debug': 10,
-    'basic': 500,
-    'confident': 5000
+    'basic': 50,
+    'confident': 200
 }
 
 # Template json file used to test scale rules
 STRESS_ACL_TABLE_TEMPLATE = "acl/templates/acltb_test_stress_acl_table.j2"
 STRESS_ACL_RULE_TEMPLATE = "acl/templates/acltb_test_stress_acl_rules.j2"
-STRESS_ACL_BASH_TEMPLATE = "acl/templates/acltb_test_stress_acl.sh"
 STRESS_ACL_TABLE_JSON_FILE = "/tmp/acltb_test_stress_acl_table.json"
 STRESS_ACL_RULE_JSON_FILE = "/tmp/acltb_test_stress_acl_rules.json"
-STRESS_ACL_BASH_FILE = "/tmp/acltb_test_stress_acl.sh"
-STRESS_ACL_BASH_LOG_FILE = "/tmp/acltb_test_stress_acl.log"
 
 LOG_EXPECT_ACL_TABLE_CREATE_RE = ".*Created ACL table.*"
 LOG_EXPECT_ACL_RULE_FAILED_RE = ".*Failed to create ACL rule.*"
 
+ACL_RULE_NUMS = 10
+
+
+@pytest.fixture(scope='module')
+def prepare_test_file(rand_selected_dut):
+    # Define a custom table type CUSTOM_TYPE by loading a json configuration
+    rand_selected_dut.template(src=STRESS_ACL_TABLE_TEMPLATE, dest=STRESS_ACL_TABLE_JSON_FILE)
+    rand_selected_dut.shell("sonic-cfggen -j {} -w".format(STRESS_ACL_TABLE_JSON_FILE))
+    # Copy acl rules
+    rand_selected_dut.template(src=STRESS_ACL_RULE_TEMPLATE, dest=STRESS_ACL_RULE_JSON_FILE)
+
 
 @pytest.fixture(scope='module')
-def setup_stress_acl_table(rand_selected_dut, tbinfo, duthost):
+def prepare_dst_ip(rand_selected_dut, tbinfo):
+    routes = namedtuple('routes', ['route', 'aspath'])
+    common_config = tbinfo['topo']['properties']['configuration_properties'].get('common', {})
+    iproute_info = get_ip_route_info(rand_selected_dut)
+    dst_prefix_list = []
+    route_prefix_len = get_route_prefix_len(tbinfo, common_config)
+    for route_prefix in iproute_info:
+        if "/{}".format(route_prefix_len) in route_prefix:
+            multipath = iproute_info[route_prefix][0].get('multipath', False)
+            if multipath:
+                out = iproute_info[route_prefix][0].get('path').split(' ')
+                aspath = out[1:]
+                entry = routes(route_prefix, ' '.join(aspath))
+                dst_prefix_list.append(entry)
+                break
+
+    route_to_send = dst_prefix_list[0].route
+    dst_ip_addr = route_to_send.strip('/{}'.format(route_prefix_len))
+
+    return dst_ip_addr
```

Comment:
Can you please clarify what's the purpose of `prepare_dst_ip`? If we are going to generate northbound traffic, the destination IP can be any value except for the vlan subnet. Similarly, if we want to generate southbound traffic, we can set the destination IP to the vlan subnet.

Reply:
Make sense, directly find northbound ports.